frontend/app/services/zuora/ZuoraService.scala

```diff
@@ -36,78 +38,72 @@ object ZuoraServiceHelpers {
 }
 
 class ZuoraService(val apiConfig: ZuoraApiConfig) extends LazyLogging {
-  import services.zuora.ZuoraServiceHelpers._
+  import ZuoraServiceHelpers._
 
   val metrics = new TouchpointBackendMetrics with StatusMetrics with AuthenticationMetrics {
     val backendEnv = apiConfig.envName
 
     val service = "Zuora"
 
-    def recordError {
+    def recordError() {
       put("error-count", 1)
     }
   }
 
-  val featuresTask = ScheduledTask(s"Zuora ${apiConfig.envName} retrieve features of membership tiers", Seq[Feature](), 0.seconds, 1.day) {
-    val featuresF = query[Feature]("Status = 'Active'")
+  private val authSupplier =
+    new FutureSupplier[Authentication](request(Login(apiConfig)))
 
-    featuresF.foreach { features =>
-      val diff = FeatureChoice.codes &~ features.map(_.code).toSet
-      lazy val msg =
-        s"""
-           |Zuora ${apiConfig.envName} is missing the following product features:
-           |${diff.mkString(", ")}. Please update configuration ASAP!"""
-          .stripMargin
+  val featuresSupplier =
+    new FutureSupplier[Seq[Feature]](getFeatures)
 
-      if (diff.nonEmpty) logger.error(msg)
-    }
+  val pingSupplier =
+    new FutureSupplier[DateTime](authRequest(
+      Query("SELECT Id FROM Product")).map { _ => new DateTime })
 
-    featuresF
-  }
+  private val scheduler = Akka.system.scheduler
 
-  lazy val featuresSchedule = featuresTask.start()
+  List(
+    (30.minutes, authSupplier),
+    (30.seconds, pingSupplier),
+    (30.minutes, featuresSupplier)
+  ) foreach { case (duration, supplier) =>
+    scheduler.schedule(duration, duration) { supplier.refresh() }
+  }
 
-  def authTaskWithCallbacks(callbacks: Seq[() => Unit]) =
-    ScheduledTask(s"Zuora ${apiConfig.envName} auth", Authentication("", ""), 0.seconds, 30.minutes) {
-      val eventualAuthentication = request(Login(apiConfig))
-      eventualAuthentication.filter(_.token.nonEmpty).foreach(_ => callbacks.foreach(_()))
-      eventualAuthentication
-    }
+  def getAuth = authSupplier.get()
 
-  val authTask = authTaskWithCallbacks(Seq(() => featuresSchedule))
+  def authRequest[T <: ZuoraResult](action: => ZuoraAction[T])(implicit reader: ZuoraReader[T]): Future[T] =
```

Comment:
The name `authRequest` read authenticat**ion** request at first, but than I figured out it was authenticat**ed** request. I would use something that suggests this better this better, ie `authenticatedRequest` or `secureRequest`